### PR TITLE
Add editable ability to FbxTextField

### PR DESCRIPTION
**Pull request by nlitwin**
### Description

Adding the ability for `FbxTextField` to be "editable". This means that it will be in `readonly` mode until you select the `Edit` button. Once you're done typing, you can select `Done`, or you have the option to rollback to the previous value by clicking the `X`.

PR Resources | Links
------ | ------
🎫 Ticket | [P1-865](https://fundbox.atlassian.net/browse/P1-865)

### Todos

- [x] Add section for displaying icomoon icons
- [x] Make input editable
- [x] Handle clicking `Done`
- [x] Handle rolling back to the previous value when clicking `X`
- [x] Handle validations
- [x] Handle working with the `currency` mask

### Preview
Note: CSS my slightly change

### CODE REVIEW

File: src/components/FbxTextField/FbxTextField.vue [hunk 1]
```diff
@@ -25,7 +27,21 @@
           @change="onChange"
         />
 
-        <span class="fbx-text-field__dollar-sign" v-if="isCurrency">$</span>
+        <span class="fbx-text-field__edit" v-if="showEdit" @click="onEdit">Edit</span>
```
Comment (forforeach) [inline src/components/FbxTextField/FbxTextField.vue]: I would wrap it with another container that uses `v-if="editable". Then, inside `v-if="isEditing"` - `v-else`. Once you do it, you don't need `showEdit` and `showDoneIcons` computed fields

File: src/components/FbxTextField/FbxTextField.spec.js [hunk 2]
```diff
@@ -8,6 +8,7 @@ describe('FbxTextField', () => {
   describe('snapshots', () => {
     it('renders the default', () => {
       const wrapper = shallowMount(FbxTextField, {
+        sync: false,
```
Comment (nlitwin) [inline src/components/FbxTextField/FbxTextField.spec.js]: It looks like adding `sync: false,` to all unit tests causes the ones that weren't passing before to pass... go figure!

File: src/components/FbxTextField/FbxTextField.vue [hunk 3]
```diff
@@ -3,8 +3,6 @@
     <label class="fbx-text-field__label">{{ label }}</label>
     <div class="fbx-text-field__wrapper">
       <div class="fbx-text-field__input-wrapper">
-        <span class="fbx-text-field__dollar-sign" v-if="isCurrency">$</span>
```
Comment (nlitwin) [inline src/components/FbxTextField/FbxTextField.vue]: I have no idea why, but when this span was up here, I could only type one letter and then the input would automatically blur... I can't for the life of me figure out why, so I moved it back down below the `input` element

File: src/components/FbxTextField/FbxTextField.vue (hunk 1 above)
Comment (nlitwin) [reply src/components/FbxTextField/FbxTextField.vue]: 👍🏼

**Pull request merged**